### DISCUSSION TITLE
Fortran runtime error using mopac-makpol on Ubuntu 22.04 lts

### DISCUSSION THREAD

**Discussion by astampfl** [openmopac/mopac#151] [Q&A] [answered]
I'm currently trying to run mopac-makpol on the first example given in the mopac 2016 manual website for Polyethylene.

I invoke the following command: mopac-makpol MAKE_Polyethylene.dat and get the following error (see output below). Can someone give me advice as how to proceed.

---
(base) anton@anton-probook:~/Documents/mopac/test$ mopac-makpol MAKE_Polyethylene.dat 
 MAKE_Polyethylene.dat

    No files of the type: 'Polyethylene' were found.

    Press return to exit
At line 431 of file /home/conda/feedstock_root/build_artifacts/mopac_1671356921589/work/src/makpol/standard.F90 (unit = 5, file = 'fort.5')
Fortran runtime error: End of file

Error termination. Backtrace:
#0  0x7f2993e4b068 in read_sf_internal
	at ../../../libgfortran/io/transfer.c:325
#1  0x7f2993e48d7d in read_default_char1
	at ../../../libgfortran/io/read.c:455
#2  0x7f2993e4c007 in formatted_transfer_scalar_read
	at ../../../libgfortran/io/transfer.c:1787
#3  0x7f2993e4cc66 in formatted_transfer
	at ../../../libgfortran/io/transfer.c:2491
#4  0x56365caaeab5 in ???
#5  0x56365caa27b9 in ???
#6  0x56365ca961d0 in ???
#7  0x7f2993829d8f in __libc_start_call_main
	at ../sysdeps/nptl/libc_start_call_main.h:58
#8  0x7f2993829e3f in __libc_start_main_impl
	at ../csu/libc-start.c:392
#9  0x56365ca96200 in ???

---

**Comment by astampfl** [comment, marked as answer]
I've worked it out myself. The way the code is written I assume is for a Windows system only and forces one to use UPPERCASE characters only for the name of the  input file. When I change to all uppercase mopac-makpol runs to completion.  I'm checking the output but there seem to be some additional pieces of information in the .mop file that I probably can live with. The way the code treats the input filename certainly needs cleaning up - personally I would get rid of most of it and just exist gracefully if the file doesn't exist.

**Reply by godotalgorithm** [reply]
MOPAC development was Windows-native for a long time, and some of the lesser-used functionality retains some I/O artifacts because of this. MAKPOL in particular still has its own separate code base, so it has not benefitted from many of the parsing improvements made to the main MOPAC program over the years. Ideally, I will eventually merge MAKPOL into the main codebase, and these artifacts will be resolved in the process. This is presently a low development priority, since MAKPOL is not used very frequently.
